### PR TITLE
sci-libs/HDF5Plugin-Zstandard: add this ebuild

### PR DESCRIPTION
**Pull request by littlewu2508**
Zstdandard compress filter plugin for hdf5
Initial commit, add ebuilds and metedata. Support Gentoo-Prefix

Package-Manager: Portage-3.0.5, Repoman-3.0.1
Signed-off-by: Yiyang Wu <xgreenlandforwyy@gmail.com>

### CODE REVIEW

File: sci-libs/HDF5Plugin-Zstandard/HDF5Plugin-Zstandard-1.0_pre20190212.ebuild [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Copyright 2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+CMAKE_MAKEFILE_GENERATOR=ninja
```
Comment (epsilon-0) [inline sci-libs/HDF5Plugin-Zstandard/HDF5Plugin-Zstandard-1.0_pre20190212.ebuild:6]: This is the default. No need to add this line.

File: sci-libs/HDF5Plugin-Zstandard/HDF5Plugin-Zstandard-1.0_pre20190212.ebuild [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+inherit cmake
+
+DESCRIPTION="ZSTD plugin for hdf5"
+HOMEPAGE="https://github.com/aparamon/HDF5Plugin-Zstandard"
+SRC_URI="https://github.com/aparamon/HDF5Plugin-Zstandard/archive/d5afdb5f04116d5c2d1a869dc9c7c0c72832b143.tar.gz -> HDF5Plugin-Zstandard.tar.gz"
```
Comment (epsilon-0) [inline sci-libs/HDF5Plugin-Zstandard/HDF5Plugin-Zstandard-1.0_pre20190212.ebuild:11]: Change this to 
`https://github.com/aparamon/HDF5Plugin-Zstandard/archive/${COMMIT}.tar.gz -> ${P}.tar.gz`
And then add a line before it saying
`COMMIT=d5afdb5f04116d5c2d1a869dc9c7c0c72832b143`

File: sci-libs/HDF5Plugin-Zstandard/HDF5Plugin-Zstandard-1.0_pre20190212.ebuild [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
```
Comment (epsilon-0) [inline sci-libs/HDF5Plugin-Zstandard/HDF5Plugin-Zstandard-1.0_pre20190212.ebuild:12]: Add a line setting setting `S`
`S=${PN}-${COMMIT}`
This allows you to get rid of the `mv` in `src_prepare`.

File: sci-libs/HDF5Plugin-Zstandard/HDF5Plugin-Zstandard-1.0_pre20190212.ebuild [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+LICENSE="Apache-2.0"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
```
Comment (epsilon-0) [inline sci-libs/HDF5Plugin-Zstandard/HDF5Plugin-Zstandard-1.0_pre20190212.ebuild:16]: Remove empty `IUSE`.

File: sci-libs/HDF5Plugin-Zstandard/HDF5Plugin-Zstandard-1.0_pre20190212.ebuild [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+DEPEND="app-arch/zstd sci-libs/hdf5"
+RDEPEND="${DEPEND}"
+BDEPEND=""
```
Comment (epsilon-0) [inline sci-libs/HDF5Plugin-Zstandard/HDF5Plugin-Zstandard-1.0_pre20190212.ebuild:20]: Remove empty `BDEPEND`

File: sci-libs/HDF5Plugin-Zstandard/HDF5Plugin-Zstandard-1.0_pre20190212.ebuild [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+src_unpack () {
```
Comment (epsilon-0) [inline sci-libs/HDF5Plugin-Zstandard/HDF5Plugin-Zstandard-1.0_pre20190212.ebuild:21]: After setting the `S` from the above comment, you can get rid of this function.

**Review by epsilon-0 — CHANGES_REQUESTED** [review]

**Comment by littlewu2508** [comment]
Thanks for these reviews! See https://github.com/gentoo/sci/pull/987/commits/57b787e5c89ef981439a5f87a0271eec536c2646 for the modification

I also changed the package version, after consulting @heroxbd

**Comment by epsilon-0** [comment]
@littlewu2508 could you squash all of this into one commit.
Also change the commit message to describe what the package is, something like:
```
sci-libs/HDF5Plugin-Zstandard: new zstd plugin for hdf5

Signed-off-by: ...
```
No need for saying `adding this ebuild`.

**Comment by littlewu2508** [comment]
> 
> 
> @littlewu2508 could you squash all of this into one commit.
> Also change the commit message to describe what the package is, something like:
> 
> ```
> sci-libs/HDF5Plugin-Zstandard: new zstd plugin for hdf5
> 
> Signed-off-by: ...
> ```
> 
> No need for saying `adding this ebuild`.

I rebased them into this commit https://github.com/gentoo/sci/commit/9b11fd4356344edae07d96d01f692341df9f4e07 with the correct commit message

**Comment by epsilon-0** [comment]
You need to regenerate the manifest.
Currently the manifest is for an incorrect version.

**Comment by littlewu2508** [comment]
> 
> 
> You need to regenerate the manifest.
> Currently the manifest is for an incorrect version.

Oops. Now the manifest is regenerated.

**Comment by epsilon-0** [comment]
merged

**Comment by epsilon-0** [comment]
Thanks for the contribution :heart:

**Pull request closed without merging**